src/helm.hoverProvider.ts

```diff
@@ -40,6 +42,15 @@ export class HelmTemplateHoverProvider implements vscode.HoverProvider {
         }
 
         if (this.notInAction(doc, pos, word)) {
+            try {
+                // have a test against whether the word is a key element in yaml
+                const { matchedNode } = yamlLocator.getYamlElement(doc, pos);
+                if (!yamlUtil.isKey(matchedNode)) {
+                    return Promise.resolve(null);
+                }
+            } catch (ex) {
+                console.log(ex);
```

Comment:
Is this desired behaviour?  As far as the user is concerned, this results in silently swallowing the exception.

Reply:
Hmm, it's a hover provider, so yeah, silence is probably best... but in that case we should be avoiding having exceptions at all.  Should we perhaps wrap this in a meaningfully named helper method (e.g. `isPositionInKey`) that internally suppresses exceptions if they are truly going to happen?

Reply:
isPositionInKey will hardly tell us with false for not in key position or exception, so I think the error wrap should be suppressed here.